### DISCUSSION TITLE
Weighted round robin

### DISCUSSION THREAD

**Discussion by shkov** [go-kit/kit#1148] [Q&A]
My `sd.Endpointer` have endpoints which call different data centers. Round Robin choses endpoints on a cyclic basis but call of some instances, placed further than other ones, takes longer time. I want the call of further endpoint only to come when near endpoint returns error. Is there way to give a priority to endpoint? If there is no realization please help me with design.

**Comment by wenkaler** [comment]
You can use this function. https://gist.github.com/wenkaler/b56305db53ffa8852b8876e3e4e5627d

**Comment by shkov** [comment]
@wenkaler Actually it solves only my problem but this approach doesn't allow to balance traffic by weight between instances. Also i am not sure that relying on endpoints order is a good practice.
